pkg/podvolume/util.go

```diff
@@ -71,14 +96,73 @@ func GetVolumeBackupsForPod(podVolumeBackups []*velerov1api.PodVolumeBackup, pod
 			continue
 		}
 
-		volumes[pvb.Spec.Volume] = pvb.Status.SnapshotID
+		volumes[pvb.Spec.Volume] = VolumeBackupInfo{
+			SnapshotID:     pvb.Status.SnapshotID,
+			UploaderType:   getUploaderTypeOrDefault(pvb.Spec.UploaderType),
+			RepositoryType: getRepositoryType(pvb.Spec.UploaderType),
+		}
 	}
 
 	if len(volumes) > 0 {
 		return volumes
 	}
 
-	return getPodSnapshotAnnotations(pod)
+	fromAnnntation := getPodSnapshotAnnotations(pod)
+	if fromAnnntation == nil {
+		return nil
+	}
+
+	for k, v := range fromAnnntation {
+		volumes[k] = VolumeBackupInfo{v, uploader.ResticType, velerov1api.BackupRepositoryTypeRestic}
+	}
+
+	return volumes
+}
+
+func GetSnapshotIdentifier(podVolumeBackups *velerov1api.PodVolumeBackupList) []repository.SnapshotIdentifier {
```

Comment:
Please add comments as it's an exported func

Reply:
Done, thanks.